src/modules/UI/Wallets/action.js

```diff
@@ -436,6 +438,30 @@ export function addNewTokenThenDeleteOldSuccess (data: any) {
   }
 }
 
+export function updateWalletLoadingProgress (walletId: string, addressLoadingProgress: number) {
+  const data = {
+    walletId,
+    addressLoadingProgress
+  }
+  return {
+    type: UPDATE_WALLET_LOADING_PROGRESS,
+    data
+  }
+}
+
+export function insertWalletIdsForProgress (activeWalletIds: Array<string>) {
```

Comment:
if you use the payload from the dispatch at the end of the login action, then this is not needed
additionally, try to keep data munging in the reducers